src/concrete/ml/onnx/ops_impl.py

```diff
@@ -946,11 +949,23 @@ def numpy_greater(
     Args:
         x (numpy.ndarray): Input tensor
         y (numpy.ndarray): Input tensor
+        **kwargs (Dict[str, Optional[Callable]]): Contains the `AutoRounder` which determines
+            the number of the least significant bits to remove
 
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
 
+    # Retrieve _auto_rounder if rounding is used
+    _auto_rounder = kwargs.get("_auto_rounder", None)
+
+    if _auto_rounder is not None:
```

Comment:
this code is duplicated several times, could you add a utility function:

```
def rounded_comparison(x, y, auto_rounder, operation: Callable):
        assert isinstance(_auto_rounder, AutoRounder)
        half = 1 << (_auto_rounder.lsbs_to_remove - 1) if _auto_rounder.is_adjusted else 0
        rounded_subtraction = round_bit_pattern((y - x) - half, lsbs_to_remove=_auto_rounder)
        return operation(rounded_subtraction)
```

then you call it like

```
 return rounded_comparison(x,y,rounder, lambda x: x >= 0)
```

Reply:
Good idea